PathPlanning/DynamicMovementPrimitives/dynamic_movement_primitives.py

```diff
@@ -0,0 +1,260 @@
+
+        t = np.arange(0, self.timesteps * self.dt, self.dt)
+        return t, np.asarray(positions)
+
+    @staticmethod
+    def dist_between(p1, p2):
+        return np.linalg.norm(p1 - p2)
+
+    def view_trajectory(self, path, title=None, demo=False):
+
+        path = np.asarray(path)
+
+        plt.cla()
+        plt.plot(self.training_data[:, 0], self.training_data[:, 1],
+                 label="Training Data")
+        plt.plot(path[:, 0], path[:, 1],
+                 linewidth=2, label="DMP Approximation")
+
+        plt.xlabel("X Position")
+        plt.ylabel("Y Position")
+        plt.legend()
+
+        if title is not None:
+            plt.title(title)
+
+        if demo:
+            plt.xlim([-0.5, 5])
+            plt.ylim([-2, 2])
+            plt.draw()
+            plt.pause(0.02)
+        else:
+            plt.show()
+
+        return t, path
+
+    def show_DMP_purpose(self):
+        """
+        This function conveys the purpose of DMPs:
+            to capture a trajectory and be able to stretch
+            and squeeze it in terms of start and stop position
+            or time
+        """
+
+        q0_orig = self.training_data[0]
+        g_orig = self.training_data[-1]
+        T_orig = self.T_orig
+
+        data_range = (np.amax(self.training_data[:, 0])
+                      - np.amin(self.training_data[:, 0])) / 4
+
+        q0_right = q0_orig + np.array([data_range, 0])
+        q0_up = q0_orig + np.array([0, data_range/2])
+        g_left = g_orig - np.array([data_range, 0])
+        g_down = g_orig - np.array([0, data_range/2])
+
+        q0_vals = np.vstack([np.linspace(q0_orig, q0_right, 20),
+                             np.linspace(q0_orig, q0_up, 20)])
+        g_vals = np.vstack([np.linspace(g_orig, g_left, 20),
+                            np.linspace(g_orig, g_down, 20)])
+        T_vals = np.linspace(T_orig, 2*T_orig, 20)
+
+        for new_q0_value in q0_vals:
+            plot_title = "Initial Position = [" + \
+                         str(round(new_q0_value[0], 2)) + ", " \
+                         + str(round(new_q0_value[1], 2)) + "]"
+            _, path = self.recreate_trajectory(new_q0_value, g_orig, T_orig)
+            self.view_trajectory(path, title=plot_title, demo=True)
+
+        for new_g_value in g_vals:
+            plot_title = "Goal Position = [" + \
+                         str(round(new_g_value[0], 2)) + ", " \
+                         + str(round(new_g_value[1], 2)) + "]"
+
+            _, path = self.recreate_trajectory(q0_orig, new_g_value, T_orig)
+            self.view_trajectory(path, title=plot_title, demo=True)
+
+        for new_T_value in T_vals:
+            plot_title = "Period = " + str(round(new_T_value, 2)) + "[sec]"
+
+            _, path = self.recreate_trajectory(q0_orig, g_orig, new_T_value)
+            self.view_trajectory(path, title=plot_title, demo=True)
+
+
+if __name__ == '__main__':
+
+    t = np.arange(0, 3*np.pi/2, 0.01)
```

Comment:
This example codes should be in a function `main` or something I think.

Reply:
Added a function example_DMP() that does everything that was originally inside this conditional